### ISSUE TITLE
[4.4] Model Export

### ISSUE THREAD

**Issue by poltergeistjoya** [poltergeistjoya/Vacant_Lot_Detection#24]
## Summary
Export trained models in production-ready format with metadata. Enable inference from saved checkpoints.

## Model Storage

```
modeling/outputs/models/nyc_2022/
  s3_rf_001.joblib                    (RF pickle)
    s3_rf_001.json                    (metadata)
  
  s3_unet_resnet34_001.pt             (U-Net weights)
    s3_unet_resnet34_001.json         (metadata)
  
  s3_deeplabv3p_resnet34_001.pt       (DeepLabV3+ weights)
    s3_deeplabv3p_resnet34_001.json   (metadata)
```

## Metadata JSON Format

```json
{
  "architecture": "unet",
  "encoder_name": "resnet34",
  "encoder_weights": "imagenet",
  "input_channels": 8,
  "training_data": {
    "train_split": "Manhattan + Staten Island",
    "val_split": "Bronx",
    "n_patches_train": 5000,
    "n_patches_val": 1500
  },
  "hyperparameters": {
    "batch_size": 16,
    "learning_rate": 1e-4,
    "optimizer": "AdamW",
    "scheduler": "CosineAnnealingLR",
    "loss_function": "BCE+Dice",
    "pos_weight": 10,
    "bce_weight": 0.5,
    "dice_weight": 0.5
  },
  "training_results": {
    "final_epoch": 42,
    "val_iou_best": 0.324,
    "val_dice_best": 0.456,
    "val_precision": 0.62,
    "val_recall": 0.38,
    "val_kappa": 0.31
  },
  "test_results": {
    "test_iou": 0.301,
    "test_dice": 0.428,
    "test_precision": 0.59,
    "test_recall": 0.35,
    "test_kappa": 0.28,
    "per_borough": {
      "brooklyn_iou": 0.295,
      "queens_iou": 0.308
    }
  },
  "training_date": "2025-03-15",
  "git_commit": "abc123def456",
  "notes": "Best model from hyperparameter sweep; soft labels applied",
  "model_size_mb": 123.4,
  "inference_time_ms_per_patch": 45
}
```

## Export Functions

```python
def save_model(model, model_path, metadata):
    '''Save PyTorch model + metadata JSON'''
    torch.save(model.state_dict(), model_path)
    with open(model_path.replace('.pt', '.json'), 'w') as f:
        json.dump(metadata, f, indent=2)

def save_rf_model(rf_clf, model_path, metadata):
    '''Save scikit-learn RF + metadata JSON'''
    import joblib
    joblib.dump(rf_clf, model_path)
    with open(model_path.replace('.joblib', '.json'), 'w') as f:
        json.dump(metadata, f, indent=2)

def load_model(model_path, architecture, device='cpu'):
    '''Load PyTorch model from checkpoint + config from metadata JSON'''
    metadata_path = model_path.replace('.pt', '.json')
    with open(metadata_path, 'r') as f:
        metadata = json.load(f)
    
    model = build_model(
        architecture, 
        encoder_name=metadata['encoder_name'], 
        in_channels=metadata['input_channels']
    )
    model.load_state_dict(torch.load(model_path, map_location=device))
    model.eval()
    return model, metadata

def load_rf_model(model_path):
    '''Load scikit-learn RF from checkpoint'''
    import joblib
    return joblib.load(model_path)
```

## Implementation Steps

1. **After Week 2/3 training:**
   - Collect final val/test metrics from TensorBoard + evaluation framework
   - Populate metadata JSON
   - Save model weights + metadata

2. **For each trained model (RF, U-Net, DeepLabV3+):**
   - Call `save_model()` or `save_rf_model()`
   - Verify file sizes (model ~130 MB for U-Net, small for RF)
   - Test `load_model()` roundtrip

3. **Archive:**
   - Compress models directory (if archiving to cloud/backup)
   - Document model lineage (which config → which results)

## Acceptance Criteria
- [ ] RF model saved as `.joblib` + `.json`
- [ ] U-Net saved as `.pt` + `.json`
- [ ] DeepLabV3+ saved as `.pt` + `.json`
- [ ] Metadata JSON complete (architecture, hyperparams, results)
- [ ] `load_model()` function works: load → eval mode → test inference
- [ ] File sizes reasonable: DL models ~130 MB, RF <10 MB
- [ ] Git commit hash recorded in metadata

## Storage & Backup
- **Local:** `modeling/outputs/models/nyc_2022/` (in repo, Git LFS if >100 MB)
- **Cloud (optional):** GCS `thesis_parcels/models/nyc_2022/` for long-term archive
- **Do not commit** model weights to main repo (use Git LFS or separate storage)

## Usage in Future
These exported models enable:
- Quick inference on new cities (load best model → run inference pipeline)
- Reproducibility (metadata documents exact hyperparams)
- Comparison baseline (for future research)

## References
- PyTorch model save/load: https://pytorch.org/tutorials/beginner/saving_loading_models.html
- sklearn joblib: https://joblib.readthedocs.io/
